### DISCUSSION TITLE
How to set the Ingress traffic to "Limited to VNet" in Azure CLI or Bicep

### DISCUSSION THREAD

**Discussion by moattarwork** [microsoft/azure-container-apps#1033] [Q&A]
I have a set of containers running inside the ManagedEnvironment with internal traffic through a VNET. I'm struggling to find a way to select "Limited To VNet" option and on every update, the ingress will change to "Limited To Container Apps Environment".
This can be done easily via UI and I was wondering if there is any way to use cli and bicep to create the same behaviour.

<img width="763" alt="Screen Shot 2022-01-31 at 8 28 13 PM" src="https://github.com/microsoft/azure-container-apps/assets/1560935/8ce18fa8-57ec-44e2-9a6e-13d7fa8e7bc5">

**Comment by ahelland** [comment]
It's not very well documented, but it is controlled by the `external` property.
false => "Limited to Container Apps Environment"
true => "Limited to vnet"
null => Ingress disabled

So, something like this in Bicep:
```
ingress: (useExternalIngress == true || useExternalIngress == false) ? {
        external: useExternalIngress
        ...
} : null
```

**Reply by PiscitelliAndrea** [reply]
I cannot find the 'external' property (I'm using version @2024-03-01, and I'm working with Bicep). 
Please indicate where you inserted it.

**Reply by visschersm** [reply]
> It's not very well documented, but it is controlled by the `external` property. false => "Limited to Container Apps Environment" true => "Limited to vnet" null => Ingress disabled
> 
> So, something like this in Bicep:
> 
> ```
> ingress: (useExternalIngress == true || useExternalIngress == false) ? {
>         external: useExternalIngress
>         ...
> } : null
> ```

So if I understand correctly the combination:
`Ingress.external: false` and 'Limited to VNET' is not a valid combination?

Strange how this is possible in the portal but not using bicep.

**Comment by vienleidl** [comment]
I managed to make it works with the command: `az containerapp ingress update -n app_name-g resource_group_name --type external --target-port 8080 --transport auto --allow-insecure true`
